### ISSUE TITLE
MQTT serverConnection lost

### ISSUE THREAD

**Issue by vamshik113** [arcoirislabs/cordova-plugin-mqtt#44]
MQTT plugin suddenly disconnects and when tried to reconnect back it shows the error:
_Error while connecting to MQTT serverConnection lost (32109) - java.net.SocketException: Software caused connection abort._ 
This happens very often. Is there any fix for this?

**Comment by adrianmihaila** [comment]
I can help you if you need it...

**Comment by vamshik113** [comment]
@adrianmihaila Yes, I'm still facing this issue. Please let me know what is the fix.

**Comment by adrianmihaila** [comment]
@vamshik113, I had the same problem, but I found a workaround. 
```
class MqttClient {
  connect(successCallback) {
    let clientId = 'm-' + device.uuid;

    cordova.plugins.CordovaMqTTPlugin.connect({
      url: MQTT_BROKER_URL,
      port: MQTT_BROKER_PORT,
      clientId: clientId,
      keepAlive: 0,
      reconnectPeriod: 1000,
      clean: false,
      success: successCallback,
      error: (e) => {
        console.error('Mqtt connection error: ' + JSON.stringify(e));
      },
      onConnectionLost: () => {
        this.connect(successCallback); // Here is the magic. On connection lost, you will try to reconnect.
      }
    });
  }
....
}

document.addEventListener('deviceready', () => {
    let mqttClient = new MqttClient();
    mqttClient.connect(() => {
          // Subscribe method is another wrapper for cordova.plugins.CordovaMqTTPlugin.subscribe()
          mqttClient.subscribe(TOPIC, QOS_LEVEL);
    });
});
```
 I made a suite of manual tests and all of them have passed.

Adi

**Comment by spilz87** [comment]
Hi

I got the same issue when two tablets with the same application connect to MQTT Server (Mosquitto on raspberry pi)

I observe that the application lost connection with server (Error while connecting to MQTT serverConnection lost (32109) - java.net.SocketException: Software caused connection abort) when on an other tablet the application connect.

They should have different Client ID 
`cordova.plugins.CordovaMqTTPlugin.connect({
  url:"tcp://" + config.gateway,
  port:config.port,
  clientId:"TABLETTE_" + macAdd,
  connectionTimeout:3000,
  willTopicConfig:{qos:0, //default is 0
    retain:true, //default is true
    topic: lastWillTopic,
    payload: lastWillPayload
  },
  username:config.MQTT_log,
  password:config.MQTT_PW,
  keepAlive:10,
  isBinaryPayload: false
});`

but I suspect an issue about that and the ClientID is the same for all application (whatever you set as clientId)

any idea ?

**Comment by arcoirislabs** [comment]
Hi,
According to MQTT spec, we cannot reuse same Client ID which is used previously. The broker shall deny connection or close previous connection. The library generates a random Client ID if it is unset or left blank.

**Comment by spilz87** [comment]
Actually if clientId is not set broker receive id null

It seems the library doesn't generate random clientId

**Comment by arcoirislabs** [comment]
A newer version is being worked on here [https://github.com/ameykshirsagar/cordova-plugin-mqtt ](Here). You can keep a tab on developments
